### ISSUE TITLE
Modifying Bundle target can result in CA certs not being available for a while

### ISSUE THREAD

**Issue by irbekrm** [cert-manager/trust-manager#100]
At the moment if  `bundle.spec.target` gets modified, trust manager will apply two updates in separate reconciles- [one where the old CA data is deleted from the target configmaps](https://github.com/cert-manager/trust-manager/blob/main/pkg/bundle/bundle.go#L116-L152), another where the new CA data is written to the target configmaps. This will happen even when the CA data does not need to change, for example if namespace selector is removed.

In practice this means that there is a period of time where the CA data is not available and in some cases it could cause issues:
- if there is a latency for calls to kube apiserver either because there is a large number of targets that trust-manager needs to update or because the apiserver is busy, the period when the CA cert is not available will be longer
- if user applies a change to the Bundle that both changes the target and also references a source that cannot be used (is an invalid CA cert), the old CA data will be removed and the configmaps will remain empty till the user fixes the issue

Assuming that we think that CA data should be 'always' available, I think it would be preferable if target updates were atomic and we didn't have a period of time where the target configmaps are empty.

I ran an experiment with 1000 namespaces and observed that the after a target namespace selector change a target configmap had its CA cert deleted and remained empty for 5 minutes till the same CA data got added back:
```
irbe@trust-manager$ k logs deploy/trust-manager -ncert-manager | grep test-1000
I0119 13:24:53.280164       1 bundle.go:151] trust/bundle "msg"="deleted old target key" "bundle"="my-org.com" "namespace"="test-1000" "old_target"={"configMap":{"key":"ca.crt"},"namespaceSelector":{"matchLabels":{"foo":"bar"}}}
I0119 13:29:15.481179       1 sync.go:230] trust/bundle "msg"="synced bundle to namespace" "bundle"="my-org.com" "namespace"="test-1000"
```

Related (wrt whether we should prioritize CA data being always available): #99

**Comment by erikgb** [comment]
I also noted this behavior when preparing a pull request for trust-manager. I wonder if there is a particular reason for doing it this way - as a two-step process? Maybe @JoshVanL has some insights?

**Comment by inteon** [comment]
As mentioned by @erikgb. This should be fixed on master thanks to #89.
